lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -497,11 +517,22 @@ Value FIRRTLTypesLowering::addArg(Type type, unsigned oldArgNumber,
   // Append the new argument.
   auto newValue = body->addArgument(type);
 
-  // Copy over the name attribute for the new argument.
+  // Remove the saved attributes for the old argument.
+  module->getAttrDictionary()
+      .getNamed(getArgAttrName(oldArgNumber))
+      .map([&](NamedAttribute attr) {
+        newAttrs.erase(attr);
+        return attr;
+      });
```

Comment:
The use of map here is clever, but in a bad way ;-) because I had to dig in to see what is going on at the API level to understand the code.  Please deal with the optional explicitly, something like:

auto argAttrs = module->getAttrDictionary().getNamed(getArgAttrName(oldArgNumber));
if (argAttrs.hasValue()) ...

Reply:
:+1: sorry, will stick to good old fashioned `if`.